config/routes.rb

```diff
@@ -1,4 +1,24 @@
 Rails.application.routes.draw do
+
+  get  "/"                                 , to: "menus#index"             , as: :all_menus
+  get  "/:menu_id"                         , to: "menus#show"              , as: :show_menu
+  get  "/:menu_id/menu_sections/new"       , to: "menu_sections#new"       , as: :new_section
+  get  "/:menu_id/preview"                 , to: "menus#preview"           , as: :preview_menu
+  get  "/:menu_id/menu_sections/:sec_id"   , to: "menu_sections#show"      , as: :show_section
+
+  post  "/"                                , to: "menus#add"
+  post "/:menu_id"                         , to: "menu_sections#add"       , as: :add_section
+  post "/:menu_id/menu_sections/:sec_id"   , to: "menu_items#add"          , as: :add_item
```

Comment:
I would the `add_item` path be:

```rb
post "/:menu_id/menu_sections/:sec_id/items"
```

Since the action is adding an item, putting `/items` on the end make it more descriptive, without that the url seems like it's posting to the sections.